src/segmantic/seg/monai_unet.py

```diff
@@ -548,3 +669,182 @@ def print_table(header, vals, indent="\t"):
                 confusion_metrics,
                 (np.squeeze(x.cpu().numpy()) for x in conf_matrix.aggregate()),
             )
+
+
+def cross_validate(
+        image_dir: Path,
+        labels_dir: Path,
+        tissue_list: Path,
+        output_dir: Path,
+        config_files_dir: Path,
+        checkpoint_file: Path = None,
+        max_epochs: int = 100,
+        generalize_test: bool = False,
+        n_splits: int = 7,
+        save_nifti: bool = True,
+        gpu_ids: List[int] = [0],
+):
+    print_config()
+    print('Cross-validating')
+    output_dir = Path(output_dir)
+    output_dir.mkdir(exist_ok=True)
+
+    tissue_dict = load_tissue_list(tissue_list)
+    print(tissue_dict)
+
+    if generalize_test:
+        generalize_test_img_path = Path(
+            "M:/DATA/ITIS/MasterThesis/Datasets/T1_T2/T1_T2_spacing_(1, 1, 1)/Hummel_Nibabel_one_only")
+        generalize_test_label_path = Path("M:/DATA/ITIS/MasterThesis/Labels/16_Labels/Hummel")
+
+        generalize_img = []
+        generalize_label = []
+
+        for file in generalize_test_img_path.iterdir():
+            generalize_img.append(file)
+        for file in generalize_test_label_path.iterdir():
+            generalize_label.append(file)
+
+        print(generalize_img)
+        print(generalize_label)
+
+        test_layers = [(16,32,64,128,256,516), (16,32,64,128), (64,128,256,512,1024)]
+        test_strides = [(2, 2, 2, 2, 2), (2, 2, 2), (2, 2, 2, 2)]
+        print('Pizza 1')
```

Comment:
I don't see why the k-fold cross-validation is mixed with the hyper-parameters. Do you always run 7-folds times all variants in your hyper-parameter matrix?

Reply:
and what does `Pizza 1` mean?